src/eos-updater-fetch.c

```diff
@@ -72,6 +74,31 @@ update_progress (OstreeAsyncProgress *progress,
     eos_updater_set_downloaded_bytes (updater, bytes);
 }
 
+static gboolean
+repo_pull (OstreeRepo *self,
+           const gchar *remote_name,
+           const gchar *ref,
+           const gchar *url_override,
+           OstreeAsyncProgress *progress,
+           GCancellable *cancellable,
+           GError **error)
+{
+  g_auto(GVariantBuilder) builder = { { { 0, } } };
+  g_autoptr(GVariant) options = NULL;
+
+  g_variant_builder_init (&builder, G_VARIANT_TYPE ("a{sv}"));
+  g_variant_builder_add (&builder, "{s@v}", "refs",
+                         g_variant_new_variant (g_variant_new_strv (&ref, 1)));
+  if (url_override != NULL)
+    g_variant_builder_add (&builder, "{s@v}", "override-url",
+                           g_variant_new_variant (g_variant_new_string (url_override)));
+
+  options = g_variant_ref_sink (g_variant_builder_end (&builder));
+  return ostree_repo_pull_with_options (self, remote_name, options,
+                                        progress, cancellable, error);
```

Comment:
When we pull from the Endless server, we rely on HTTPS to guarantee we're we can trust the content. For that reason, we don't bother to setup `gpg-verify-summary=true` in the remote configuration. Maybe we should do that anyway (flatpak does), but in the case of LAN or USB updates, I think it's very important to do that since ostree won't otherwise validate the summary file and we'll have a broken chain of trust.

You can actually pass that in as a boolean variant option to `ostree_repo_pull_with_options`. The option is called `gpg-verify-summary`. I think the thing to do is always pass that in as `TRUE` unless the config explicitly disables it. Something like:
```
gboolean gpg_verify_summary;
/* Get gpg-verify-summary setting, defaulting to TRUE */
if (!ostree_repo_get_remote_boolean_option (repo, remote_name, "gpg-verify-summary", TRUE, &gpg_verify_summary, error)
    /* Handle error reading configuration */
g_variant_builder_add (&builder, "{s@v}", "gpg-verify-summary", g_variant_new_variant (g_variant_new_boolean (gpg_verify_summary)));
```